lib/david/resource_discovery.rb

```diff
@@ -16,11 +19,37 @@ def _call(env)
 
       @env = env
 
-      filtered = routes_hash.select { |link| filter(link) }
-      body     = filtered.keys.map(&:to_s).join(',')
+      queries = @env[QUERY_STRING].split('&')
```

Comment:
Could the filtering by `rt` not be done in the `filter` method?

Reply:
At first, I tried to do exactly that.  (see: https://github.com/AnimaGUS-minerva/david/commit/d463c9dbc9905d5ea683bc942d6485daa9d6ae4c)  The problem is that one then has complex logic to decide how many times to return a resource which may be duplicated by others, so I changed my mind.

Reply:
Can't we dedup them after filtering?

Reply:
I thought about dedup, and I tried that, but it just seemed like a backwards way to do things. Can you explain why you think it's better to do this in filter?  It seems like really we are doing two different things in this query, and why conflate them?   
I also am thinking that if we moved the resource_discovery *out* of david and somehow into the application, that it would be easier to support Sinatra.